### PR TITLE
Add model, endpoints and Experiment fields for Projects

### PR DESCRIPTION
**Pull request by taylorjacklespriggs**

### CODE REVIEW

File: src/main/java/com/sigopt/model/Project.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+    public Integer getUpdated() {
+        return Utils.asInteger(this.get("updated"));
+    }
+
+    public Metadata getMetadata() {
+        return Utils.mergeInto(new Metadata(), this.get("metadata"));
+    }
+
+    public static class Builder extends APIObjectBuilder<Project> {
```
Comment (alexandraj777) [inline src/main/java/com/sigopt/model/Project.java]: Do you want metadata here?

Comment (taylorjacklespriggs) [reply src/main/java/com/sigopt/model/Project.java]: ah yes I do

**Comment by taylorjacklespriggs** [comment]
Tried these changes out in jshell

File: src/main/java/com/sigopt/model/Project.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+        public Builder name(String name) {
+            this.obj.set("name", name);
+            return this;
+        }
+
+        public Builder metadata(Map<String, String> metadata) {
+            this.obj.set("metadata", metadata);
+            return this;
+        }
```
Comment (pfhayes) [inline src/main/java/com/sigopt/model/Project.java]: looks like you're missing some of the builder fields

File: src/main/java/com/sigopt/model/Client.java [hunk 3]
```diff
@@ -48,10 +48,53 @@ public Experiments(String prefix) {
...
+                Project.class
+            );
+        }
+
+        public APIMethodCaller<Project> create() {
+            return new APIMethodCaller<Project>("post", this.prefix() + "/projects", Project.class);
+        }
+    }
+
+    public static class ProjectsResource extends BoundObject {
```
Comment (pfhayes) [inline src/main/java/com/sigopt/model/Client.java]: do these need to be separate classes?

Comment (taylorjacklespriggs) [reply src/main/java/com/sigopt/model/Client.java]: They don't absolutely need to be, but this prevents `new Client(X).projects(Y).create()`, `new Client(X).projects(Y).list()`, `new Client(X).projects().experiments()` and `new Client(X).projects().fetch()` from creating exceptions at runtime (they will be caught at compile time)

File: src/main/java/com/sigopt/model/Project.java (hunk 2 above)
Comment (taylorjacklespriggs) [reply src/main/java/com/sigopt/model/Project.java]: for project create/project update these are the only fields (therefore they are the only fields required by the builder)

**Pull request merged**